.github/workflows/continuous_integration.yml

```diff
@@ -43,5 +43,22 @@ jobs:
         run: |
           python3 -m pip install --upgrade pip
           python3 -m pip install -e ".[all]"
+      - name: Start a server
+        run: |
+          make docker
+          mkdir -p models
+          curl -L https://huggingface.co/bartowski/Llama-3.2-1B-Instruct-GGUF/resolve/main/Llama-3.2-1B-Instruct-Q5_K_S.gguf \
+          -o models/Llama-3.2-1B-Instruct-Q5_K_S.gguf
+          docker run -d \
+            -v ./models:/app/models \
+            -v ./examples/settings.json:/app/settings.json \
+            --cap-add SYS_RESOURCE \
+            -p 8000:8000 \
+            -e CONFIG_FILE=settings.json \
+            --name nekko-api-container \
+            nekko-api
       - name: Run tests
-        run: python3 -m pytest --full-trace -v
+        run: |
+          python3 -m pytest --full-trace -v
+          docker stop nekko-api-container
```

Comment:
Container cleanup code should be run even if tests fail. It may be best to stop and remove the container in a separate step in conjunction with `if: ${{ always() }}`.